usecases/indexes/index_editor.go

```diff
@@ -135,3 +138,75 @@ func (editor ClientDbIndexEditor) ListAllUniqueIndexes(ctx context.Context) ([]m
 	}
 	return editor.ingestedDataIndexesRepository.ListAllUniqueIndexes(ctx, db)
 }
+
+func (editor ClientDbIndexEditor) CreateUniqueIndexAsync(
+	ctx context.Context,
+	index models.UnicityIndex,
+) error {
+	logger := utils.LoggerFromContext(ctx)
+
+	organizationId, err := editor.organizationIdOfContext()
+	if err != nil {
+		return err
+	}
+	db, err := editor.executorFactory.NewClientDbExecutor(ctx, organizationId)
+	if err != nil {
+		return errors.Wrap(
+			err,
+			"Error while creating client schema executor in CreateUniqueIndexAsync")
+	}
+	err = editor.ingestedDataIndexesRepository.CreateUniqueIndexAsync(ctx, db, index)
+	if err != nil {
+		return errors.Wrap(err, "Error while creating unique index in CreateUniqueIndexAsync")
+	}
+	logger.InfoContext(
+		ctx,
+		fmt.Sprintf("Unique index pending creation asynchronously: %+v\n", index),
+		"org_id", organizationId,
+	)
+	return nil
+}
+
+func (editor ClientDbIndexEditor) CreateUniqueIndex(
+	ctx context.Context,
+	exec repositories.Executor,
+	index models.UnicityIndex,
+) error {
+	logger := utils.LoggerFromContext(ctx)
+	if exec == nil {
+		organizationId, err := editor.organizationIdOfContext()
+		if err != nil {
+			return err
+		}
+		exec, err = editor.executorFactory.NewClientDbExecutor(ctx, organizationId)
+		if err != nil {
+			return errors.Wrap(
+				err,
+				"Error while creating client schema executor in CreateUniqueIndex")
+		}
+	}
+
+	if err := editor.ingestedDataIndexesRepository.CreateUniqueIndex(ctx, exec, index); err != nil {
+		return errors.Wrap(err, "Error while creating unique index in CreateUniqueIndex")
+	}
+
+	logger.InfoContext(ctx, fmt.Sprintf("Unique index pending created: %+v\n", index))
+	return nil
+}
+
+func (editor ClientDbIndexEditor) DeleteUniqueIndex(
+	ctx context.Context,
+	index models.UnicityIndex,
+) error {
+	organizationId, err := editor.organizationIdOfContext()
+	if err != nil {
+		return err
+	}
+	db, err := editor.executorFactory.NewClientDbExecutor(ctx, organizationId)
+	if err != nil {
+		return errors.Wrap(
+			err,
+			"Error while creating client schema executor in DeleteUniqueIndex")
+	}
+	return editor.ingestedDataIndexesRepository.DeleteUniqueIndex(ctx, db, index)
+}
```

Comment:
Hmm, I see your point. The idea here was that it can be used in different contexts. Similarly to how there is a scenario_fetcher usecase that is used in other usecases (never directly by the api handlers) and that does not enforce permissions. That plus the fact that in practice it's all quite specific to the data model usecase - I rather separated it from it in order to keep it unit-testable (if you have a zillion repository calls to mock to test a usecase, it doesn't work any more).
That being said, you could say there is a stronger case in favor or security checks inside, since it does actions not just reading.